src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -41,6 +41,20 @@
 #define TRANSIENT_FAILURE_ATTEMPTS 3 /* Arbitrary limit: how many attempts are considered enough to report
                                       * transient failure. */
 
+/* RFC2131 section 4.1 specifies retransmission delays, starting at 4s then doubling
+ * each attempt with max of 64s, with -1 to +1 sec of random 'fuzz' added */
+#define RFC2131_FUZZ_USEC ((int32_t)(random_u32() % (2 * USEC_PER_SEC)) - (int32_t)USEC_PER_SEC)
+#define RFC2131_SEC41_NEXT_TIMEOUT_USEC(now, attempt) \
+        ((usec_t)(now + ((int32_t)((UINT64_C(1) << MIN((attempt) + 1, (uint64_t) 6)) * USEC_PER_SEC) + RFC2131_FUZZ_USEC)))
```

Comment:
please place `now` in `()` to avoid evaluation order issues

Reply:
sorry missed that, corrected